src/veins/modules/mobility/traci/subscriptionManagement/PersonSubscriptionManager.h

```diff
@@ -0,0 +1,153 @@
+
+public:
+    /**
+     * Constructor.
+     */
+    PersonSubscriptionManager();
+
+    /**
+     * Default destructor.
+     */
+    virtual ~PersonSubscriptionManager() = default;
+
+    /**
+     * Update this PersonSubscriptionManager with the given
+     * active person id list. You can force an update of active
+     * persons using this.
+     *
+     * @param currentlyActivePersonIds A list containing the currently active persons
+     * identified by their ids.
+     *
+     * @return true
+     */
+    bool updateWithList(std::list<std::string>& currentlyActivePersonIds);
+
+    /**
+     * Update this manager with the given buffer containing a subscription
+     * response for a person value.
+     *
+     * This will create new subscriptions for specific persons if the buffer
+     * contained an ID_LIST subscription.
+     *
+     * @param buffer the traci buffer containing the subscription response.
+     *
+     * @return bool true if the given buffer contained an ID_LIST response.
+     */
+    bool update(TraCIBuffer& buffer) override;
+
+    /**
+     * This gives you all the persons that were updated (includes
+     * new persons) since the last time you called this method. That
+     * means that multiple calls to this method will NOT yield the same
+     * result.
+     *
+     * This can also contain multiple updates for a single person.
+     *
+     * @return a list of TraCIPerson.
+     */
+    std::list<TraCIPerson> getUpdated();
+
+    /**
+     * This gives you all the disappeared persons since you called
+     * this method the last time. That means that multiple calls
+     * to this method will yield different results.
+     *
+     * @return std::set<std::string> a set of person ids.
+     */
+    std::set<std::string> getDisappeared();
+
+    /**
+     * Initialize this manager with the given parameters to access TraCI.
+     */
+    void initialize(std::shared_ptr<TraCIConnection> connection, std::shared_ptr<TraCICommandInterface> commandInterface) override;
+
+private:
+
+    /**
+     * This contains all updated persons. Will be cleared after
+     * getUpdated() was called.
+     */
+    std::list<TraCIPerson> mUpdatedPersons;
+
+    /**
+     * This is the difference between mActivePersonIds and
+     * currently active persons (provided with call to update()).
+     *
+     * This takes all the elements which are contained in
+     * mActivePersons but not in currently active person ids.
+     */
+    std::set<std::string> mDisappearedPersons;
+
+    /**
+     * Get the active person id set.
+     *
+     * @return a set of string containing the active person ids.
+     */
+    std::set<std::string> getActivePersonIds();
+
+    /**
+     * Process a subscription result that contains an id list. This is
+     * simply a helper method to improve clarity of update().
+     */
+    void processPersonIDList(std::list<std::string>& idList);
+
+    /**
+     * Subscribe to specific TraCI person variables.
```

Comment:
What are those specific variables? Can they be changed or read somewhere?
E.g., the variables could be put into a `public const static std::vector` of TraCI Variables.

Reply:
variable names added. They are all defined in TraCIConstants.h header file.

Reply:
Alright.